core/src/stage.cpp

```diff
@@ -112,28 +112,6 @@ InterfaceFlags StagePrivate::interfaceFlags() const {
 	return f;
 }
 
-void StagePrivate::pruneInterface(InterfaceFlags accepted) {
-	const InterfaceFlags accepted_start{ accepted & START_IF_MASK };
-	const InterfaceFlags accepted_end{ accepted & END_IF_MASK };
-
-	const InterfaceFlags required{ requiredInterface() };
-	const InterfaceFlags required_start{ required & START_IF_MASK };
-	const InterfaceFlags required_end{ required & END_IF_MASK };
-
-	if (required == UNKNOWN)
-		throw std::logic_error("stage type does not specify interface, but also does not override pruneInterface. "
-		                       "Who sets up its interfaces?");
-
-	// only one side needs to be specified but the value has to be compatible with this stage
-	if (((accepted_start != UNKNOWN) && (accepted_start & required_start) != required_start) ||
-	    ((accepted_end != UNKNOWN) && (accepted_end & required_end) != required_end)) {
-		boost::format desc("interface %1% %2% does not match inferred interface %3% %4%");
-		desc % flowSymbol<START_IF_MASK>(required_start) % flowSymbol<END_IF_MASK>(required_end);
-		desc % flowSymbol<START_IF_MASK>(accepted_start) % flowSymbol<END_IF_MASK>(accepted_end);
-		throw InitStageException(*me(), desc.str());
-	}
-}
-
```

Comment:
Why did you remove the check?
I don't see harm in testing all stages during setup time, although of course the invariant *should* hold for simple interfaces.